### PR TITLE
Update SteamScript: fix "content servers unreachable"

### PR DESCRIPTION
**Pull request by Zemogiter**
Fixing #663 
The fix is not applied probably due to the script not being able to detect when the Steam client is closed. This fix cannot be applied if Steam client is running. This also makes the script fail to create a shortcut after shuting down Steam.

### CODE REVIEW

**Comment by plata** [comment]
What you did is: you defined a function within the `go` function and never call it (so the Codacy complain is correct).
You should instead:
1. Define the function e.g. in l. 72 as `SteamScript.prototype.fixCertificateIssue = function (wine) {`
2. Call it (in the place where you defined it now): `this.fixCertificateIssue(wine);`

**Comment by Zemogiter** [comment]
After the changes the script gives me this error after I finish Steam instalation (with the option "Launch Steam" unchecked):
```
javax.script.ScriptException: TypeError: this.fixCertificateIssue is not a function in [engines, wine, quick_script, steam_script] at line number 116
	at jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:470)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:454)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:406)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:402)
	at jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:155)
	at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: [engines, wine, quick_script, steam_script]:116 TypeError: this.fixCertificateIssue is not a function
	at jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)
	at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:213)
	at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:185)
	at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:172)
	at jdk.nashorn.internal.runtime.Undefined.lookup(Undefined.java:102)
	at jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(NashornLinker.java:106)
	at jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(NashornLinker.java:98)
	at jdk.internal.dynalink.support.CompositeTypeBasedGuardingDynamicLinker.getGuardedInvocation(CompositeTypeBasedGuardingDynamicLinker.java:176)
	at jdk.internal.dynalink.support.CompositeGuardingDynamicLinker.getGuardedInvocation(CompositeGuardingDynamicLinker.java:124)
	at jdk.internal.dynalink.support.LinkerServicesImpl.getGuardedInvocation(LinkerServicesImpl.java:154)
	at jdk.internal.dynalink.DynamicLinker.relink(DynamicLinker.java:253)
	at jdk.nashorn.internal.scripts.Script$Recompilation$253$5260$\^eval\_.go([engines, wine, quick_script, steam_script]:116)
	at jdk.nashorn.internal.scripts.Script$Recompilation$243$275$\^eval\_.run(<eval>:10)
	at org.phoenicis.scripts.Installer$$NashornJavaAdapter.run(Unknown Source)
	at jdk.nashorn.internal.scripts.Script$217$\^eval\_.:program(<eval>:28)
	at jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:637)
	at jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494)
	at jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:449)
	... 10 more
```

File: Engines/Wine/QuickScript/Steam Script/script.js [hunk 1]
```diff
@@ -70,6 +70,15 @@ SteamScript.prototype.downloadFinished = function (wine) {
     }
 };
 
+SteamScript.prototype.FixCertificateIssue = function (wine){
```
Comment (plata) [inline Engines/Wine/QuickScript/Steam Script/script.js]: This must be `fixCertificateIssue` (it's case sensitive).

File: Engines/Wine/QuickScript/Steam Script/script.js [hunk 2]
```diff
@@ -103,6 +112,8 @@ SteamScript.prototype.go = function () {
 
     wine.run(tempFile, [], null, false, true);
 
+    this.fixCertificateIssue(wine);
```
Comment (plata) [inline Engines/Wine/QuickScript/Steam Script/script.js]: Please add a comment here why this is required (something like "fix Wine bug 12345 (cannot download games)").

**Comment by Zemogiter** [comment]
Downloading the games form steam works. But we need to add a message for the user that tells them to shutdown Steam once they logged in. Only then the `FixCertificateIssue` is applied. And there is another problem. Even after installation of the steam game is complete and the Steam client is turned off the script is still waiting for the download to finish.

**Comment by Zemogiter** [comment]
Yes but its not displayed.

**Comment by plata** [comment]
Ok... That's strange. I'll have to take a look at that.

Do you think the fix itself is stable enough to be merged?

**Comment by Zemogiter** [comment]
Without that message displaying correctly in the wizard, we shouldn't merge this yet because some users my just uncheck "Run Steam" after instalation which will prevent the creation of config.vdf file. This file MUST exist before the code for the fix is executed.
There might be something wrong with this part of the SteamScript:

` setupWizard.wait(tr("Please follow the steps of the Steam setup.\n\nUncheck \"Run Steam\" or close Steam completely after the setup so that the installation of \"{0}\" can continue.", this._name));`

because it does not show up during installation at all. Once we fix it we also have to change it to something like this:

"Please close Steam client completely after you log in and pass Steam Guard. Otherwise the script will fail to install the \"{0}\" "

File: Engines/Wine/QuickScript/Steam Script/script.js [hunk 3]
```diff
@@ -69,6 +69,15 @@ SteamScript.prototype.downloadFinished = function (wine) {
         return false;
     }
 };
+// Fix for the "content server unavaiable" error
+SteamScript.prototype.FixCertificateIssue = function (wine){
```
Comment (plata) [inline Engines/Wine/QuickScript/Steam Script/script.js]: function name should be lower camel case (`fixCertificateIssue`)

File: Engines/Wine/QuickScript/Steam Script/script.js [hunk 4]
```diff
@@ -94,14 +103,15 @@ SteamScript.prototype.go = function () {
...
         .get();
 
-    setupWizard.wait(tr("Please follow the steps of the Steam setup.\n\nUncheck \"Run Steam\" or close Steam completely after the setup so that the installation of \"{0}\" can continue.", this._name));
-
     var wine = new Wine()
         .wizard(setupWizard)
         .prefix(this._name, this._wineDistribution, this._wineArchitecture, this._wineVersion)
         .luna();
+    
+    setupWizard.wait(tr("Please follow the steps of the Steam setup. After updating and loging in, shut down completly Steam client so the installation of \"{0}\" can continue.", this._name));
```
Comment (plata) [inline Engines/Wine/QuickScript/Steam Script/script.js]: I would prefer:
"Please follow the steps of the Steam setup. Then, wait until Steam is updated, log in and finally close Steam completely so the installation of ..."

**Comment by plata** [comment]
I have merged the close message fix to master so please merge master to your branch.

You should also add comments why logging in and changing the config is required. After that I think we can merge this.

**Comment by Zemogiter** [comment]
Ready for review.

**Pull request merged**